x-pack/plugins/ml/public/application/components/saved_objects_warning/saved_objects_warning.tsx

```diff
@@ -5,48 +5,69 @@
  * 2.0.
  */
 
-import React, { FC, useEffect, useState } from 'react';
-
+import React, { FC, useEffect, useState, useCallback, useRef, useMemo } from 'react';
 import { EuiCallOut, EuiLink, EuiSpacer } from '@elastic/eui';
 import { FormattedMessage } from '@kbn/i18n/react';
 import { JobType } from '../../../../common/types/saved_objects';
-import { useMlApiContext, useMlKibana } from '../../contexts/kibana';
+import { useMlApiContext } from '../../contexts/kibana';
+import { JobSpacesSyncFlyout } from '../../components/job_spaces_sync';
+import { checkPermission } from '../../capabilities/check_capabilities';
 
 interface Props {
   jobType?: JobType;
+  onCloseFlyout?: () => void;
+  forceRefresh?: boolean;
 }
 
-export const SavedObjectsWarning: FC<Props> = ({ jobType }) => {
+export const SavedObjectsWarning: FC<Props> = ({ jobType, onCloseFlyout, forceRefresh }) => {
   const {
-    savedObjects: { initSavedObjects },
+    savedObjects: { syncCheck },
   } = useMlApiContext();
-  const {
-    services: {
-      http: { basePath },
-    },
-  } = useMlKibana();
 
+  const mounted = useRef(false);
   const [showWarning, setShowWarning] = useState(false);
+  const [showSyncFlyout, setShowSyncFlyout] = useState(false);
+  const canCreateJob = useMemo(() => checkPermission('canCreateJob'), []);
 
-  useEffect(() => {
-    let unmounted = false;
-    initSavedObjects(true)
-      .then(({ jobs }) => {
-        if (unmounted === true) {
-          return;
-        }
+  const checkStatus = useCallback(async () => {
+    try {
+      if (mounted.current === false) {
+        return;
+      }
+
+      const { result } = await syncCheck(jobType);
 
-        const missingJobs =
-          jobs.length > 0 && (jobType === undefined || jobs.some(({ type }) => type === jobType));
-        setShowWarning(missingJobs);
-      })
-      .catch(() => {
-        console.log('Saved object synchronization check could not be performed.'); // eslint-disable-line no-console
-      });
+      setShowWarning(showSyncFlyout || result);
+    } catch (error) {
+      console.log('Saved object synchronization check could not be performed.'); // eslint-disable-line no-console
+    }
+  }, [showSyncFlyout, setShowWarning]);
+
+  useEffect(() => {
+    mounted.current = true;
+    if (forceRefresh === undefined || forceRefresh === true) {
+      checkStatus();
+    }
     return () => {
-      unmounted = true;
+      mounted.current = false;
     };
-  }, []);
+  }, [forceRefresh, mounted]);
+
+  const onClose = useCallback(() => {
+    if (forceRefresh === undefined) {
+      checkStatus();
+    }
+    setShowSyncFlyout(false);
+    if (typeof onCloseFlyout === 'function') {
+      onCloseFlyout();
+    }
+  }, [checkStatus, onCloseFlyout, setShowSyncFlyout]);
+
+  useEffect(() => {
```

Comment:
```suggestion
  useEffect(funtion hideSyncFlyoutOnWarningClose() {
```
or something like that